frontend/src/components/user-administration/user-administration-modal/user-administration-modal.tsx

```diff
@@ -0,0 +1,278 @@
+import { zodResolver } from '@hookform/resolvers/zod';
+import {
+  Box,
+  Button,
+  FormControl,
+  Grid,
+  InputLabel,
+  MenuItem,
+  Modal,
+  Select,
+  SelectChangeEvent,
+  Typography
+} from '@mui/material';
+import React, { useState } from 'react';
+import { TextFieldElement, useForm } from 'react-hook-form-mui';
+import { toast } from 'react-toastify';
+import { z, ZodIssueOptionalMessage, ErrorMapCtx } from 'zod';
+import { makeStyles } from '@mui/styles';
+import { Icon } from '@iconify/react';
+import { User } from 'pages/api/user';
+import ShowPasswordModal from './show-password-modal';
+import { UserDto } from 'src/models/user/user.model';
+
+const UserAdministrationModal = ({
+  showModal = true,
+  setShowModal,
+  onAddedUser,
+  user
+}: {
+  showModal: boolean;
+  setShowModal: (value: boolean) => void;
+  onAddedUser: (value: any) => void;
```

Comment:
You can put the userModel here instead of any
```suggestion
  onAddedUser: (value: UserDto) => void;
```